Extractor/Config/ConfigToggles.cs

```diff
@@ -0,0 +1,58 @@
+namespace Cognite.OpcUa.Config
+{
+    /// <summary>
+    /// Class wrapping a config instance, enabling features that may be enabled from different places.
+    /// </summary>
+    public class ConfigToggles
+    {
+        private readonly FullConfig config;
+        public ConfigToggles(FullConfig config)
+        {
+            this.config = config;
+        }
+
+        private bool FdmEnabled => config.Cognite?.MetadataTargets?.DataModels?.Enabled ?? false;
+
+        /// <summary>
+        /// Should the extractor load references to and from properties?
+        /// </summary>
+        public bool MapPropertyReferences => FdmEnabled;
+
+        /// <summary>
+        /// Should the extractor get references to types when loading non-hierarchical references?
+        /// </summary>
+        public bool GetNonHierarchicalTypeReferences => FdmEnabled || config.Extraction.NodeTypes.AsNodes;
+        /// <summary>
+        /// Should the extractor create unmapped nodes it encounters when reading non-hierarchical references, or just ignore them?
+        /// </summary>
+        public bool CreateUnknownReferencedNodes => FdmEnabled || config.Extraction.Relationships.CreateReferencedNodes;
+        /// <summary>
+        /// Should the extractor create nodes with NodeClass ObjectType, ReferenceType, VariableType, or DataType?
```

Comment:
This sounds like the bool makes a distinction between 4 different cases, can we improve the comment here?

Reply:
They are toggles, each enable a feature independently.

Reply:
I improved the docs a bit